support/src/main/java/com/yubico/yubikit/support/Device.java

```diff
@@ -0,0 +1,490 @@
+/*
+ * Copyright (C) 2022 Yubico.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.yubico.yubikit.support;
+
+import static com.yubico.yubikit.management.Capability.FIDO2;
+import static com.yubico.yubikit.management.Capability.OATH;
+import static com.yubico.yubikit.management.Capability.OPENPGP;
+import static com.yubico.yubikit.management.Capability.OTP;
+import static com.yubico.yubikit.management.Capability.PIV;
+import static com.yubico.yubikit.management.Capability.U2F;
+
+import com.yubico.yubikit.core.Logger;
+import com.yubico.yubikit.core.Transport;
+import com.yubico.yubikit.core.Version;
+import com.yubico.yubikit.core.YubiKeyConnection;
+import com.yubico.yubikit.core.application.ApplicationNotAvailableException;
+import com.yubico.yubikit.core.application.CommandException;
+import com.yubico.yubikit.core.fido.FidoConnection;
+import com.yubico.yubikit.core.otp.OtpConnection;
+import com.yubico.yubikit.core.smartcard.SmartCardConnection;
+import com.yubico.yubikit.core.smartcard.SmartCardProtocol;
+import com.yubico.yubikit.core.util.Pair;
+import com.yubico.yubikit.management.Capability;
+import com.yubico.yubikit.management.DeviceConfig;
+import com.yubico.yubikit.management.DeviceInfo;
+import com.yubico.yubikit.management.FormFactor;
+import com.yubico.yubikit.management.ManagementSession;
+import com.yubico.yubikit.management.UsbInterface;
+import com.yubico.yubikit.yubiotp.YubiOtpSession;
+
+import java.io.IOException;
+import java.security.InvalidParameterException;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.EnumMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.StringJoiner;
+
+import javax.annotation.Nonnull;
+import javax.annotation.Nullable;
+
+public class Device {
```

Comment:
I don't really like the name "Device" here, but I'm not sure what to suggest as a replacement... `DeviceInfoUtil` or `DeviceUtil` maybe?